### ISSUE TITLE
CLI Hook not firing

### ISSUE THREAD

**Issue by FabienTaillon** [forcedotcom/cli#578]
### Summary

I tried to use the new [CLI Hook](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_hooks.htm) mechanism, both in a custom plugin and with the [plugin-metadata-hook-demo](https://github.com/salesforcecli/plugin-metadata-hook-demo) but nothing is firing.

### Steps To Reproduce:

I tried both to:
* Create a custom plugin with a hook as explained [here](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_hooks_steps.htm)
* Use the [plugin-metadata-hook-demo](https://github.com/salesforcecli/plugin-metadata-hook-demo) plugin, doing all the required steps (yarn install, prepack, plugins:link, and checking with sfdx plugins). Nothing happens, metadata isn't change, no console log is displayed. Also, while debugging, I can debug steps by steps but when adding breakpoints in the CLI plugins none of them is reached. 

### Expected result

The hook is firing.

### Actual result

The hook isn't firing.

### Additional information

@oclif/plugin-autocomplete 0.2.0
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.0 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/lwc-dev-server 2.5.1
├─ @oclif/plugin-help 2.2.1
└─ @oclif/plugin-update 1.3.9
@salesforce/sfdx-trust 3.2.1 (core)
analytics 1.12.0 (core)
generator 1.1.3 (core)
plugin-metadata-hook-demo 48.8.0 (link) /Users/fabien/Documents/code/plugin-metadata-hook-demo
salesforcedx 49.7.0 (core)
├─ salesforcedx-templates 49.4.0 (core)
├─ custom-metadata 1.0.8 (core)
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
└─ salesforce-alm 49.7.0 (core)
sfdx-cli 7.71.0 (core)

Latest stable macOS version (10.15.5)

node v12.14.1

**Comment by FabienTaillon** [comment]
I tried to install the latest LTS version of nodejs (v12.18.3) [as recommended](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli.htm) just in case, but doesn't change anything.

**Comment by FabienTaillon** [comment]
Just updated to this week's release and now it works without me changing anything.

salesforcedx 49.9.2 (core)
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
├─ salesforce-alm 49.10.0 (core)
├─ apex 0.0.9 (core)
├─ custom-metadata 1.0.10 (core)
└─ templates 49.4.4 (core)
sfdx-cli 7.73.2 (core)

**Comment by FabienTaillon** [comment]
Reopening this one as it's not working again with the latest version of the CLI.
If I downgrade manually to salesforce-alm@49.11.0 it's working again.

salesforcedx 49.13.1 (core)
├─ apex 0.1.1 (core)
├─ custom-metadata 1.0.10 (core)
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
├─ salesforce-alm 49.12.2 (core)
└─ templates 49.9.1 (core)
sfdx-cli 7.75.1 (core)

**Comment by amphro** [comment]
We'll have to take a look. We did a fix for hooks not firing on failed deploys, and as of then, they were working well. In QA'ing that item, we did find a problem with multiple versions of the `@salesforce/core` library. [This should fix it](https://github.com/forcedotcom/sfdx-core/pull/310) and should be in v50 of the CLI being released this weekend. If there are any previous versions of core being loaded into memory though, it is possible that the problem may still persist for a while unless a clean install. There is no way for us to force that.
